sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelFactory.java

```diff
@@ -103,4 +103,9 @@
 
   /** Converts the given string to a {@link Path}. */
   Path toPath(String path);
+
+  /**
+   * Delete files with the given {@code uris} if they exist.
+   */
+  void deleteIfExists(Collection<String> uris) throws IOException;
```

Comment:
Generally we should provide a tightly scoped primitive operation in `IOChannelFactory` and feel free to provide utilities around it - so this can be `delete` and throw a `FileNotFoundException` if the underlying IO cannot delete it, and then we can reuse the "try delete and suppress `FileNotFoundExceptions`